alvr/server/src/connection.rs

```diff
@@ -569,10 +569,26 @@ fn connection_pipeline(
                     microphone_desc.devices.clone(),
                 )
                 .to_con()?;
-                if alvr_audio::is_same_device(&game_audio_device, &sink)
-                    || alvr_audio::is_same_device(&game_audio_device, &source)
+                match &settings
+                    .audio
+                    .microphone
+                    .as_option()
+                    .map(|config: &alvr_session::MicrophoneConfig| config.devices.clone())
+                    .unwrap()
                 {
-                    con_bail!("Game audio and microphone cannot point to the same device!");
+                    alvr_session::MicrophoneDevicesConfig::VBCable => {
```

Comment:
No need to get the microphone config again as it's already bound to `microphone_desc`.
I think this could be rewritten as:
```rust
if matches!(microphone_desc.devices, alvr_session::MicrophoneDevicesConfig::VBCable) {
   ...
}
```
In general, try to avoid using unwrap.

You should also rename `microphone_desc` to `microphone_config` because that got missed in a previous refactoring.

Reply:
There is no refactoring pending. That was a while ago. You can rename now.

Reply:
> There is no refactoring pending. That was a while ago. You can rename now.

Sorry, I am not quite sure what you mean. I can see the variable `microphone_desc` still available in the code in the master branch but not for `microphone_config`.

Reply:
That's because it's your job to rename the variable there won't be a *_config till you rename it in this pr

Reply:
Oh, I see. Sorry about that. I misunderstood your words on how it is being "missed". Now, it is been updated.

Reply:
Well they weren't my words but I understand